packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,43 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number | string) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
+      if (!isNaN(val)) {
```

Comment:
I don't think it really matters here since you first coerce it into a number, but I read online that `Number.isNaN` is more reliable than the global `isNaN`, so I was thinking we might as well use `Number.isNaN` instead

Reply:
Oh very interesting! Done!